test/units/testsuite-75.sh

```diff
@@ -293,6 +295,20 @@ run getent -s myhostname hosts localhost
 grep -qE "^127\.0\.0\.1\s+localhost" "$RUN_OUT"
 enable_ipv6
 
+# Issue: https://github.com/systemd/systemd/issues/25088
+run getent -s resolve hosts 127.128.0.5
+grep -qEx '127\.128\.0\.5\s+localhost5(\s+localhost5?\.localdomain[45]?){4}' "$RUN_OUT"
+[ "$(wc -l <"$RUN_OUT")" -eq 1 ]
+
+# Issue: https://github.com/systemd/systemd/issues/20158
+run dig +noall +answer +additional localhost5.
+grep -qEx 'localhost5\.\s+0\s+IN\s+A\s+127\.128\.0\.5' "$RUN_OUT"
+[ "$(wc -l <"$RUN_OUT")" -eq 1 ]
+run dig +noall +answer +additional localhost5.localdomain4.
+grep -qEx 'localhost5\.localdomain4\.\s+0\s+IN\s+CNAME\s+localhost5\.' "$RUN_OUT"
+grep -qEx 'localhost5\.\s+0\s+IN\s+A\s+127\.128\.0\.5' "$RUN_OUT"
+[ "$(wc -l <"$RUN_OUT")" -eq 2 ]
+
 : "--- Basic resolved tests ---"
```

Comment:
This would have revealed the fix is not finished.
```
run dig +short -x 127.0.0.5
[ "$(wc -l <"$RUN_OUT")" -eq 1 ]
```

Reply:
If `run dig +short -x 127.128.0.5` printed just one line instead of five, it would have broken the following test above:
```
run getent -s resolve hosts 127.128.0.5
grep -qEx '127\.128\.0\.5\s+localhost5(\s+localhost5?\.localdomain[45]?){4}' "$RUN_OUT"
```

Reply:
No. dig only talks DNS protocol over port 53. ``getent -s resolve`` talks to by NSS API, which can be served a bit different record format. systemd-resolved has all needed information, it just needs to use it.

Reply:
Apparently, libnss_resolve also talks to systemd-resolved using DNS protocol, so AFAIK there is no different record format available to libnss_resolve. Do you have a better idea for what systemd-resolved could produce using DNS protocol so that libnss_resolve would be able to serve /etc/hosts as expected?